plugin/storage/cassandra/spanstore/operation_names_test.go

```diff
@@ -118,28 +139,40 @@ func TestOperationNamesStorageGetServices(t *testing.T) {
 		return true
 	})
 	matchEverything := mock.MatchedBy(func(v []interface{}) bool { return true })
-	for _, expErr := range []error{nil, scanError} {
-		withOperationNamesStorage(writeCacheTTL, func(s *operationNameStorageTest) {
+	for _, test := range []struct {
+		name          string
+		schemaVersion string
+		expErr        error
+	}{
+		{name: "test old schema without error", schemaVersion: previousVersion, expErr: nil},
+		{name: "test old schema with scan error", schemaVersion: previousVersion, expErr: scanError},
+		{name: "test new schema without error", schemaVersion: latestVersion, expErr: nil},
+		{name: "test new schema with scan error", schemaVersion: latestVersion, expErr: scanError},
+	} {
+		fmt.Printf(test.name)
+		withOperationNamesStorage(0, test.schemaVersion, func(s *operationNameStorageTest) {
 			iter := &mocks.Iterator{}
 			iter.On("Scan", matchOnce).Return(true)
 			iter.On("Scan", matchEverything).Return(false) // false to stop the loop
-			iter.On("Close").Return(expErr)
+			iter.On("Close").Return(test.expErr)
 
 			query := &mocks.Query{}
 			query.On("Iter").Return(iter)
 
-			s.session.On("Query", mock.AnythingOfType("string"), []interface{}{"service-a"}).Return(query)
-
+			s.session.On("Query", mock.AnythingOfType("string"), mock.Anything).Return(query)
 			services, err := s.storage.GetOperations("service-a")
-			if expErr == nil {
+			if test.expErr == nil {
 				assert.NoError(t, err)
-				// expect empty string because mock iter.Scan(&placeholder) does not write to `placeholder`
-				assert.Equal(t, []string{""}, services)
+				if test.schemaVersion == previousVersion {
+					// expect one empty operation result because mock iter.Scan(&placeholder) does not write to `placeholder`
+					assert.Equal(t, []string{""}, services)
+				} else {
+					assert.Equal(t, []string{}, services)
+				}
 			} else {
-				assert.EqualError(t, err, "Error reading operation_names from storage: "+expErr.Error())
+				assert.EqualError(t, err, fmt.Sprintf("Error reading %s from storage: %s", schemas[test.schemaVersion].tableName, test.expErr.Error()))
```

Comment:
arguments to assert functions are already treated as printf args, you don't need additional fmt.Sprintf

Reply:
I tried, but it failed. 

`func EqualError(t TestingT, theError error, errString string, msgAndArgs ...interface{}) bool`
fmt.Sprintf is used to generate the errString, without fmt.Sprintf, it will treat "schemas[test.schemaVersion].tableName, test.expErr.Error()" as msg args for the output when the assert fail